api/src/main/java/javax/application/converter/AbstractPrimitiveNumberConverter.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package javax.application.converter;
+
+/**
+ * @author Andres Almiray
+ */
+public abstract class AbstractPrimitiveNumberConverter<T extends Number> extends AbstractPrimitiveConverter<T> implements NumberConverter<T> {
+    protected abstract Class<T> getTypeClass();
```

Comment:
protected abstract Class<T> getTypeClass();

Can we get this directly from class meta data?

Reply:
No. Unless we had TypeLiteral support in core JDK, which we don't.

Reply:
It's not, but I would take a closer look at CDI 2 (which now also has a Java SE only part)

Reply:
isn't CDI "too big"?

Reply:
It depends what you mean by big. If it takes most of the pain from your back and give you feature set that can reduce your work it is never big enough.

If does not add value or it adds very little then it will always be "to big".

I was thinking about doing an implementation with CDI 2 SE as base and all annotations supported as CDI extensions.

Reply:
I meant too big to be run on a mobile platform. You need a container that can provide you the basics, but nothing more. How would you get a stripped down version of a CDI impl? Also, Even Guice (the RI for JSR-330) doesn't run out of the box on Android, you must include a `no-aop` version of the dependency. CDI *might* work on the desktop.